mw-webapp/src/pages/authPage/AuthPage.tsx

```diff
@@ -1,14 +1,36 @@
-import {Link} from "react-router-dom";
-import {useAuth} from "src/utils/useAuth";
+import {User} from "firebase/auth";
+import {useEffect, useState} from "react";
+import {Link, useNavigate} from "react-router-dom";
+import {Button} from "src/component/button/Button";
+import {getNewUserCredentials} from "src/utils/auth/getNewUserCredentials";
+import {handleUserAuthState} from "src/utils/auth/handleAuthState";
+import {handleLogIn} from "src/utils/auth/handleLogIn";
+import {handleLogOut} from "src/utils/auth/handleLogOut";
 
 export const AuthPage = () => {
-  const handleGoogleSignIn = useAuth();
+  const [user, setUser] = useState<User | null>(null);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    return () => {
+      handleUserAuthState(setUser);
+      getNewUserCredentials(navigate, "/main");
+    };
+  }, []);
 
   return (
     <>
-      <button onClick={handleGoogleSignIn}>
-        Sign in with Google
-      </button>
+      {user ? (
```

Comment:
dont use ternarian operator in multiline expressions if possibe. Maybe in this case it will be more appropriate to render the same buttom with different props depeds on does user exist

Reply:
Done